src/test/routes/deleteDefinition.spec.ts

```diff
@@ -0,0 +1,75 @@
+import { appTest } from "../../main/app.test";
+import { expect } from "chai";
+import * as idamServiceMock from "../http-mocks/idam";
+import * as mock from "nock";
+import * as request from "supertest";
+
+describe("Confirm Delete page", () => {
+    describe("on POST /deletedefinition", () => {
+
+        it("should redirect to the Confirm Delete page when Yes or No is not chosen", () => {
+            idamServiceMock.resolveRetrieveUserFor("1", "admin");
+            idamServiceMock.resolveRetrieveServiceToken();
+
+            return request(appTest)
+                .post("/deletedefinition")
+                .send({ definitionVersion: 1, itemToDelete: "definition", jurisdictionId: "TEST" })
+                .set("Cookie", "accessToken=ey123.ey456")
+                .then((res) => {
+                    expect(res.statusCode).to.equal(302);
+                    expect(res.headers.location.startsWith(
+                      "/deleteitem?item=definition&jurisdictionId=TEST&version=1")).to.be.true;
+                });
+        });
+        it("should redirect to the Definitions list when No is chosen", () => {
+            idamServiceMock.resolveRetrieveUserFor("1", "admin");
+            idamServiceMock.resolveRetrieveServiceToken();
+
+            return request(appTest)
+                .post("/deletedefinition")
+                .send({ definitionVersion: 1, deleteItem: "No", itemToDelete: "definition", jurisdictionId: "TEST" })
+                .set("Cookie", "accessToken=ey123.ey456")
+                .then((res) => {
+                    expect(res.statusCode).to.equal(302);
+                    expect(res.headers.location.startsWith("/definitions")).to.be.true;
+                });
+        });
+
+        it("should redirect to the Definitions list when Yes is chosen", () => {
+            idamServiceMock.resolveRetrieveUserFor("1", "admin");
+            idamServiceMock.resolveRetrieveServiceToken();
+
+            mock("http://localhost:4451")
+                .delete("/api/draft/TEST/1")
+                .reply(204);
+
+            return request(appTest)
+                .post("/deletedefinition")
+                .send({ definitionVersion: 1, deleteItem: "Yes", itemToDelete: "definition", jurisdictionId: "TEST" })
+                .set("Cookie", "accessToken=ey123.ey456")
+                .then((res) => {
+                    expect(res.statusCode).to.equal(302);
+                    expect(res.headers.location.startsWith("/definitions")).to.be.true;
+                });
+        });
+
+        it("should redirect to the Definitions list when Yes is chosen but an error occurred", () => {
+            idamServiceMock.resolveRetrieveUserFor("1", "admin");
+            idamServiceMock.resolveRetrieveServiceToken();
+
+            mock("http://localhost:4451")
```

Comment:
A few places you call nock as nock and this one is mock.  Which one we shall use to make the whole damn thing consistent

Reply:
Good point - I'll rename to whichever is used more widely :)

Reply:
There does seem to be some consistency... for whatever reason, `nock` is imported as "mock" in all the `routes` tests, but as "nock" in all the `service` tests. Therefore, I'm going to leave this one as is.